lib/src/util/number.dart

```diff
@@ -118,3 +119,46 @@ double moduloLikeSass(double num1, double num2) {
   var result = num1 % num2;
   return result == 0 ? 0 : result + num2;
 }
+
+/// Return square root of [number]
+SassNumber sqrt(SassNumber number) {
+  number.assertNoUnits();
+  return SassNumber(math.sqrt(number.value));
+}
+
+/// Return [num1]^[num2]
+SassNumber pow(SassNumber num1, SassNumber num2) {
+  num1.assertNoUnits();
+  num2.assertNoUnits();
+  return SassNumber(math.pow(num1.value, num2.value));
+}
+
+/// Return the rounded number according to the strategy
+SassNumber roundStrategies(SassString strategy, SassNumber number) {
+  if (!number.value.isNaN && number.value.isFinite) {
+    var strategies = {
+      'nearest': SassNumber(number.value.round().toDouble()),
+      'up': SassNumber(number.value.ceil().toDouble()),
+      'down': SassNumber(number.value.floor().toDouble()),
+      'to-zero': number.value < 0
+          ? SassNumber(number.value.ceil().toDouble())
+          : SassNumber(number.value.floor().toDouble())
+    };
+
+    return strategies[strategy.text] ?? SassNumber(double.nan);
+  }
+  return SassNumber(double.nan);
+}
+
+/// Return the rounded number according to the strategy and the step provided.
+SassNumber step(SassString strategy, SassNumber number, SassNumber step) {
+  var stepStrategies = {
```

Comment:
Here (and above), this evaluates each possible strategy every time. Since `stepStrategies` is not actually passed around, this should just be an `if` or `switch` statement.

If we did need to actually pass around some sort of unified representation of each strategy's implementation, we could wrap them in a function to prevent unnecessary calculation.

Reply:
Done! I moved the logic to the `calculation.dart` file and used a switch :)